### ISSUE TITLE
TabPane should pass different classNames to NavItem and tab-pane

### ISSUE THREAD

**Issue by kennyw12** [react-bootstrap/react-bootstrap#1054]
We would like to style the NavItem and tab-pane div within TabPane differently by passing in different classNames. 

This functionality currently isn't supported by passing in className on a TabPane, since a TabPane className prop is passed to both the TabPane and NavItem.
https://github.com/react-bootstrap/react-bootstrap/blob/master/src/TabbedArea.js#L85

```
<TabPane navClassName="my-nav-item" paneClassName="my-tab-pane" {...myotherProps}>My Tab Pane</TabPane>
```

Would you be open to a PR?

/cc @stubbornella @gpleiss @matt-royal @ctaymor

**Comment by stubbornella** [comment]
To give broader context, we are trying to be as consistent as we can about how className, id, and style attributes are passed into an element so that folks who want to modify components can anticipate this behavior. 

What we've settled on so far is that ```className```, ```id```, and ```style``` are passed to the outer-most wrapper element. ```InnerClassName``` is being applied to the inner element of a container. So, for example, on a panel, innerClassName gets passed through to the `.panel-body` element.

Tabs are a little more complicated because there are multiple tab panes, so we had thought that classNames applied to individual tabs would apply to only those tab-panes (not the tab itself), and innerClassName would apply to all tab-panes (we'd merge classNames). 

I'm realizing this is a little hard to explain in words... ;) we're happy to jump on a hangout or just do a PR so you can see what we are suggesting in code!

**Comment by jquense** [comment]
bah yeah TabbedArea is a leaky component right now...I am having a bit of a hard time following what you folks are looking for, but it seems that passing the tabpane className to the navItem is causing the issue? We are definitely open to a PR, not sure if trying to maintain compatibility here makes sense...the current behavior of passing the className to both the navItem and the tabPane strike me as odd, and possibly a bug.

@react-bootstrap/collaborators anyone know _why_ this logic is here in the first place, it seems likes its always been there?

**Comment by mtscout6** [comment]
The Tab component is another component in dire need of some love. If you're open to doing some work on it to clean it so the external API is clean and intuitive that would be great. This is right in line with our goals for our 1.0.0 release.

**Comment by kennyw12** [comment]
We are happy to help with the design with this component. Are there any issues your team has encountered that we can keep in mind when considering changes to the API? We are already planning to review open PRs and issues regarding TabbedArea and TabPane.

**Comment by mtscout6** [comment]
Well, since you're taking feature requests :wink: If you can do a review to ensure that it meet the W3C ARIA Specs for Accessibility that would be awesome too. For details see: http://www.w3.org/WAI/PF/aria-practices/#tabpanel

**Comment by jquense** [comment]
hehe I was just going to add that, I did a recent pass over the TabbedArea component for ARIA compliance but its still pretty kludgy. It could definitely use some love.

**Comment by stubbornella** [comment]
Jason, what did you think could be improved? We need to make all of our components accessible, but we're still learning about what that means.




Matt, we'll check out those docs this week.



Thanks!

**Comment by jquense** [comment]
Right now in order to tie tabs to to particular panes via the `aria-describedby` and `aria-controls` the component generates Id's passed off of the `props.id` of the TabbedArea. I had to do things that aren't super elegant, like expose a `linkID` prop on navitem in order to set it. It works, but it is potentially collision prone, and isn't customizeable at all. It works fine given the current api, where the Pane is doing most of the work for defining the navItems. It all might only need to change if ya'll are changing the api substantially. Feel free to ping me if something doesn't make sense

**Comment by mtscout6** [comment]
Essentially Accessible means that the component can be used by persons of various disabilities. Usually they do so with some form of assisted technology. For example blind folks may use a screen reader that will convert text to audio for them to hear. Blind people will also often be restricted to only using the keyboard to access everything since they cannot see the mouse on the screen. On the flip side you may have a paraplegic that is using a trackball in his mouth which acts as a mouse and they don't use the keyboard. [WebAim](http://webaim.org/) has some good material to explain the concept and factors to keep in mind when developing your sites. The key takeaway though is that by making your site accessible you will also make it friendlier for the rest of your audience as well. For example as a society we started making buttons bigger because they were hard to tap on our phones with such a small screen. This invariably helped people with Parkinson's disease to use a mouse easier since their target area to click was larger. The Web Standards Committee W3C has put together the WAI spec specifically to help developers to achieve such needs. That's what those `aria-*` properties are on html elements. This is especially important if you ever write software for U.S. and other National government sites since many countries will have laws restricting the use of software due to these requirements.

**Comment by kennyw12** [comment]
We tried reworking the api.
https://github.com/react-bootstrap/react-bootstrap/pull/1091

We will be creating separate PRs for Accessibility and a few other features.

**Comment by taion** [comment]
Closed by #1091 (unless I'm mistaken).
